### PR TITLE
MNTOR-1923: Monitor premium sub fxa event handler

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-1923

# Description
* On `subscribe`
  * activate OneRep profile
  * start opt-out process 
* On `unsubscribe`
  * deactivate OneRep profile (auto stops opt-out)

# Screenshot (if applicable)

Not applicable.

# How to test

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 1]
```diff
@@ -57,7 +64,9 @@ const getJwtPubKey = async () => {
  * @param {NextRequest} req
  * @returns {Promise<jwt.JwtPayload>} decoded JWT data, which should contain FxA events
  */
-const authenticateFxaJWT = async (req: NextRequest) => {
+const authenticateFxaJWT = async (
```
Comment (pdehaan) [inline src/app/api/v1/fxa-rp-events/route.ts:67]: nit: (although prior art) Should this be "JWT" or "Jwt" to be consistent w/ L39 above (and the `JwtPayload` references)?

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 2]
```diff
@@ -228,6 +237,29 @@ export async function POST(request: NextRequest) {
           event,
           updatedSubscriptionFromEvent,
         });
+
+        // get profile id
+        const result = await getOnerepProfileId(subscriber);
+        const oneRepProfileId = result[0]["onerep_profile_id"] as number;
```
Comment (pdehaan) [inline src/app/api/v1/fxa-rp-events/route.ts:243]: Q: Do we know if `result` array will have at least one item here?
```js
async function getOnerepProfileId (subscriberId) {
  return await knex('subscribers')
    .select('onerep_profile_id')
    .where('id', subscriberId)
}
```

File: src/app/functions/server/onerep.ts [hunk 3]
```diff
@@ -154,6 +154,23 @@ export async function activateProfile(profileId: number): Promise<void> {
...
+      method: "PUT",
+    }
+  );
+  if (!response.ok) {
+    log.info(
+      `Failed to deactivate OneRep profile: [${response.status}] [${response.statusText}]`
+    );
+    throw new Error(
+      `Failed to deactivate OneRep profile: [${response.status}] [${response.statusText}]`
+    );
```
Comment (pdehaan) [inline src/app/functions/server/onerep.ts:170]: I think we use this pattern a bunch of places already, so more of a random question; but does it make sense to move the duplicated message into a temp var and use it for the logging and error message?

```suggestion
    const msg = `Failed to deactivate OneRep profile: [${response.status}] [${response.statusText}]`;
    log.info(msg);
    throw new Error(msg);
```

File: src/app/api/v1/fxa-rp-events/route.ts (hunk 1 above)
Comment (mansaj) [reply src/app/api/v1/fxa-rp-events/route.ts]: fixed thanks

File: src/app/api/v1/fxa-rp-events/route.ts (hunk 2 above)
Comment (mansaj) [reply src/app/api/v1/fxa-rp-events/route.ts]: added optional chaining, thanks

**Comment by mansaj** [comment]
@rhelmer if we aren't doing the UI

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 4]
```diff
@@ -228,6 +237,29 @@ export async function POST(request: NextRequest) {
           event,
           updatedSubscriptionFromEvent,
         });
+
+        // get profile id
```
Comment (rhelmer) [inline src/app/api/v1/fxa-rp-events/route.ts:241]: I created an `activateProfile` in https://github.com/mozilla/blurts-server/pull/3307 if you want to use that, we could land that first? I think for Foxfooding we need to do that approach anyway.

File: src/app/functions/server/onerep.ts [hunk 5]
```diff
@@ -154,6 +154,23 @@ export async function activateProfile(profileId: number): Promise<void> {
   }
 }
 
+export async function deactivateProfile(profileId: number): Promise<void> {
```
Comment (rhelmer) [inline src/app/functions/server/onerep.ts:157]: Same as above, https://github.com/mozilla/blurts-server/pull/3307 implements this if you want to use it (it's basically identical to yours)

File: src/app/functions/server/onerep.ts (hunk 3 above)
Comment (rhelmer) [reply src/app/functions/server/onerep.ts:170]: Maybe we should consider an error function?

**Review by rhelmer — APPROVED** [review]
lgtm, but I'd create an `activateProfile` (you can take from https://github.com/mozilla/blurts-server/pull/3307). You can land this first and I'll reconcile, or wait for that one to land, up to you.

File: src/app/api/v1/fxa-rp-events/route.ts (hunk 4 above)
Comment (mansaj) [reply src/app/api/v1/fxa-rp-events/route.ts:241]: `activateProfile` existed before?

**Comment by mansaj** [comment]
@rhelmer Since this and #3307 are so similiar, perhaps the best way is to just close this PR and copy paste the `route.ts` changes into your branch

**Pull request closed without merging**